features/fixtures/mazeracer/app/src/main/java/com/bugsnag/mazeracer/scenarios/OkhttpAutoInstrumentNetworkCallbackScenario.kt

```diff
@@ -0,0 +1,58 @@
+package com.bugsnag.mazeracer.scenarios
+
+import android.util.Log
+import com.bugsnag.android.performance.BugsnagPerformance
+import com.bugsnag.android.performance.NetworkRequestInstrumentationCallback
+import com.bugsnag.android.performance.PerformanceConfiguration
+import com.bugsnag.android.performance.internal.InternalDebug
+import com.bugsnag.android.performance.okhttp.BugsnagPerformanceOkhttp
+import com.bugsnag.mazeracer.Scenario
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import kotlin.concurrent.thread
+
+class OkhttpAutoInstrumentNetworkCallbackScenario(
+    config: PerformanceConfiguration,
+    scenarioMetadata: String,
+) : Scenario(config, scenarioMetadata) {
+    init {
+        InternalDebug.spanBatchSizeSendTriggerPoint = 1
+    }
+
+    override fun startScenario() {
+        config.networkRequestCallback = NetworkRequestInstrumentationCallback { reqInfo ->
+            reqInfo.url?.let { url ->
+                if (url.startsWith("http://bs-local.com")) {
+                    reqInfo.url = null
+                } else if (url == "https://google.com/") {
+                    reqInfo.url = null
+                } else if (url.endsWith("/changeme")) {
+                    reqInfo.url = url.dropLast(9) + "/changed"
+                }
+            }
```

Comment:
purely stylistic nitpicking 😄, but we could drop an indent level by using an Elvis early-exit:

```suggestion
            val url = reqInfo.url ?: return@NetworkRequestInstrumentationCallback
            if (url.startsWith("http://bs-local.com")) {
                reqInfo.url = null
            } else if (url == "https://google.com/") {
                reqInfo.url = null
            } else if (url.endsWith("/changeme")) {
                reqInfo.url = url.dropLast(9) + "/changed"
            }
```